### PR TITLE
Qingwei/fix trades

### PR DESCRIPTION
**Pull request by qingweibinary**

### CODE REVIEW

File: src/_store/AppStateProvider.js [hunk 1]
```diff
@@ -10,6 +10,18 @@ import { appStateSelector } from '../_selectors/AppStateSelectors';
 @connect(appStateSelector)
 export default class AppStateProvider extends Component {
 
+    componentWillMount() {
+        this.setState({ timer: window.setTimeout(this.showMessageForSlowConnection.bind(this), 4000) });
+    }
+
+    componentWillUnmount() {
+        window.clearTimeout(this.state.timer);
```
Comment (borisyankov) [inline src/_store/AppStateProvider.js]: No need for state. This from clock component:

componentDidMount() {
        this.interval = setInterval(() =>
            this.setState({ time: Date.now() }), 1000);
    }

    componentWillUnmount() {
		clearInterval(this.interval);
	}

Comment (qingweibinary) [inline src/_store/AppStateProvider.js]: Did not thought of ~ will fix

**Comment by qingweibinary** [comment]
@borisyankov have a look, if you really do not like the Contract Object, I could revert it, but then the UI will blink a little bit

**Comment by borisyankov** [comment]
Good enough ;)

**Pull request merged**